functions.js

```diff
@@ -1 +1,112 @@
-// YOUR CODE HERE
+// ### maxOrMin(num1, num2, max)
+// Define a function maxOrMin that takes three parameters: two numbers and a
+// boolean. Have it return the larger of the two numbers if the boolean is true,
+// otherwise have it return the lesser of the numbers.
+
+function maxOrMin(a, b, boolean) {
+  // if boolean is true & a is greater than b, return a
+  if ( (boolean) && (a > b) ) {
+    return a;
+  // if boolean is true & b is greater than a, return b
+  } if ( (boolean) && (b > a) ) {
+    return b;
+  // if boolean is not true & a is less than b, return a
+  } else if ( (!boolean) && (a < b) ) {
+    return a;
+  // if boolean is not true & b is less than a, return b
+  } else if ( (!boolean) && (b < a) ) {
+    return b;
+  }
+}
+
+maxOrMin(1, 2, "");
+
+// ### `sillySum(arr)`
+// Write a function that takes an array of numbers and returns the sum of each number multiplied by its index.
+
+// count += (number * index)
+// Solution 1
+var sillyArray = [2, 4, 5, 6];
+var result = 0;
+//start at 0, stop after array goes beyond array length, iterate by +1
+//input element and multiply by its index value
+function sillySum(array) {
+for (i = 0; i < sillyArray.length; i++) {
+  result += i * sillyArray[i];
+}
+
+console.log(result);
+
+  }
+};
+
+// Solution 2
+var sillyArray = [1, 6, 2, 7, 9]
+
+function sillySum(arr){
+		return arr.map(function callback(element, index) {
+			return element + (element * index)
+		});
+}
+
+sillySum(sillyArray);
+
+// take array
+// return sum of each element's (number * index)
+// prime number = num not divisible by any number except 1 or itself
+
+// ### numSquare(max)
+// Create a function called numSquare that will return an array of all perfect
+// square numbers up to, but not exceeding a max number.
+// Perfect squares: 4, 9, 16, 25, 36, 49, 64, 81
+//
```

Comment:
functions should be declared using this syntax:
function numSquare(x, y){
  ...
}